scripts/transform.sh

```diff
@@ -85,11 +90,16 @@ function set_doi() {
     echo "${article_id#*10.1101/}"
 }
 
-function handle_stdin() {
-    local input="$(cat /dev/stdin)"
-
-    if [[ "${input: -1}" != $'\n' ]]; then
-        input="${input}"$'\n'
+function handle_input_xml() {
+    if [[ -n "$INPUT_XML_FILE" && -n "$(echo "$INPUT_XML_FILE" | tr -d '[:space:]')" ]]; then
+        # read from variable
+        local input="$(cat "$INPUT_XML_FILE")"
+    else
+        # read from stdin
+        local input="$(cat /dev/stdin)"
+        if [[ "${input: -1}" != $'\n' ]]; then
+            input="${input}"$'\n'
+        fi
```

Comment:
Just a slight adjustment of this script allows the xml to be supplied as stdin or the location of a file can be provided. Storing in a file and referencing the file simplified the process of handling large inputs from the api request.